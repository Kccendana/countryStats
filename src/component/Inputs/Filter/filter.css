.filter {
  background-color: white;
  color: #b13967;
  padding: 0.5em;
  border-radius: 8px;
  border-top-right-radius: 8px;
}

input.filter-input {
  border: none;
  color: #b13967;
  font-size: 0.9em;
}

.dropdown {
  position: absolute;
  background-color: white;
  color: #b13967;
  width: 205px;
  padding: 0.5em;
  text-align: left;
  border-bottom-left-radius: 8px;
  top: 46px;
  border-bottom-right-radius: 8px;
}
