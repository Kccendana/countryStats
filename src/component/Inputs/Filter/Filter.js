import { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleDown } from '@fortawesome/free-solid-svg-icons';
import { setRegion } from '../../../redux/countrySlice';

import './filter.css';

const Filter = () => {
  const regions = ['Africa', 'Oceania', 'Americas', 'Asia', 'Europe'];
  const [filter, setFilter] = useState('');
  const [displayDropdown, setDisplayDropdown] = useState(false);

  const dispatch = useDispatch();

  const handleDropdown = () => {
    setDisplayDropdown(!displayDropdown);
  };

  useEffect(() => {
    if (filter !== '') {
      dispatch(setRegion(filter.toLocaleLowerCase()));
    }
  }, [dispatch, filter]);
  return (
    <section className="filter-container">
      <div className="filter" role="button" onClick={handleDropdown} onKeyDown={handleDropdown} tabIndex={-1}>
        <input
          type="text"
          readOnly
          placeholder="Filter by Region"
          value={filter}
          className="filter-input"
        />

        <FontAwesomeIcon icon={faAngleDown} />
      </div>

      {displayDropdown ? (
        <div className="dropdown">
          {regions.map((item) => (
            <div
              key={item}
              className="dropdown-item"
              onClick={() => {
                setFilter(item);
                handleDropdown();
              }}
              onKeyDown={() => {
                setFilter(item);
                handleDropdown();
              }}
              tabIndex={-1}
              role="button"
            >
              {item}
            </div>
          ))}
        </div>
      ) : null}
    </section>
  );
};

export default Filter;
