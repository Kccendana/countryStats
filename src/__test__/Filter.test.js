import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Filter from '../component/Inputs/Filter/Filter';

jest.mock('axios', () => ({
  get: jest.fn(),
}));

describe('renders Filter component', () => {
  it('matches the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Filter />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
