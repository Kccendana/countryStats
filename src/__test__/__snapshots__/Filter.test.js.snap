// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Filter component matches the snapshot 1`] = `
<div>
  <section
    class="filter-container"
  >
    <div
      class="filter"
    >
      <input
        class="filter-input"
        placeholder="Filter by Region"
        readonly=""
        type="text"
        value=""
      />
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-angle-down "
        data-icon="angle-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M201.4 342.6c12.5 12.5 32.8 12.5 45.3 0l160-160c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L224 274.7 86.6 137.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l160 160z"
          fill="currentColor"
        />
      </svg>
    </div>
  </section>
</div>
`;
